mitosheet/mitosheet/api/graph/plotly_express_graphs.py

```diff
@@ -0,0 +1,230 @@
+from typing import Any, Dict, List, Union
+import pandas as pd
+from mitosheet.api.graph.graph_parameter import GraphParameter
+from mitosheet.transpiler.transpile_utils import column_header_list_to_transpiled_code, column_header_to_transpiled_code
+from mitosheet.types import ColumnHeader
+import plotly.express as px
+import plotly.graph_objects as go
+from mitosheet.api.graph.graph_utils import BAR, BOX, HISTOGRAM, SCATTER, filter_df_to_safe_size, get_barmode, get_graph_title
+
+# TAB is used in place of \t in generated code because 
+# Jupyter turns \t into a grey arrow, but converts four spaces into a tab.
+TAB = '    '
+
+def graph_filtering(graph_type: str, df: pd.DataFrame, column_headers: List[ColumnHeader]) -> pd.DataFrame:
+    """
+    Filters the dataframe about to be graphed so that we don't crash the browser
+    """
+    return filter_df_to_safe_size(graph_type, df, column_headers)
+
+
+def graph_filtering_code(graph_type: str, df_name: str, df: pd.DataFrame, column_headers: List[ColumnHeader]) -> str:
+    """
+    Returns the code for filtering the dataframe so we don't crash the browser
+    """
+    # Check if we filter the graph
+    _, filtered = filter_df_to_safe_size(graph_type, df, column_headers)
+    
+    # If we don't filter the graph, then return an empty string
+    if not filtered: 
+        return ''
+    
+    # If we do filter the graph, then return the code needed to filter the graph
+    filtered_code = """from mitosheet import filter_df_to_safe_size
+
+# Filter the dataframe so that it does not crash the browser
+{df_name}_filtered, _ = filter_df_to_safe_size('{graph_type}', {df_name}, {column_headers})
+""".format(
+    graph_type=graph_type,
+    df_name=df_name, 
+    column_headers=column_headers
+)
+    return filtered_code
+
+
+def graph_creation(
+    graph_type: str, 
+    df: pd.DataFrame, 
+    x_axis_column_headers: List[ColumnHeader],
+    y_axis_column_headers: List[ColumnHeader]
+) -> go.Figure: 
+    """
+    Creates and returns the Plotly express graph figure
+    """
+    
+    # Create the parameters that we use to construct the graph
+    x_arg: Union[List[ColumnHeader], ColumnHeader, None] = None
```

Comment:
The reason we need to case more specifically is because if both the parameters are a list, then plotly express gives this error. `ValueError: Cannot accept list of column references or list of columns for both `x` and `y`.`

I'll add that to a comment though.